src/main/js/diff/Diff.tsx

```diff
@@ -167,31 +150,57 @@ const Diff: FC<Props> = ({
     if (annotations.length > 0) {
       return [<FileComments>{annotations}</FileComments>];
     }
+
     return [];
   };
 
   const annotationFactory = (context: AnnotationFactoryContext) => {
     const annotations: { [key: string]: React.ReactNode } = {};
-
     const hunkId = createHunkId(context);
-    const hunkState = diffState.lines[hunkId];
-    if (hunkState) {
-      Object.keys(hunkState).forEach((changeId: string) => {
-        const lineState = hunkState[changeId];
-        if (lineState) {
-          const lineAnnotations = [];
-          if (lineState.comments && lineState.comments.length > 0) {
-            lineAnnotations.push(createComments(lineState.comments));
-          }
-          if (lineState.editor) {
-            lineAnnotations.push(createNewCommentEditor(lineState.location));
-          }
-          if (lineAnnotations.length > 0) {
-            annotations[changeId] = <InlineComments>{lineAnnotations}</InlineComments>;
+    const commentsByLine: { [key: string]: Comment[] } = {};
+
+    comments?._embedded.pullRequestComments.forEach(comment => {
+      if (comment.location?.hunk) {
+        const hunkLocation = createHunkIdFromLocation(comment.location);
+        if (comment.location?.hunk && isInlineLocation(comment.location) && hunkLocation === hunkId) {
+          const changeId = createChangeIdFromLocation(comment.location);
+          let lineComments = commentsByLine[changeId];
+          if (!lineComments) {
+            lineComments = [];
+            commentsByLine[changeId] = lineComments;
           }
+          lineComments.push(comment);
         }
-      });
-    }
+      }
+    });
```

Comment:
Using streams, I would use `filter` instead of `if`s, though it's not really nice here:
```javascript
    comments?._embedded.pullRequestComments
      .filter(comment => comment.location?.hunk)
      .filter(
        comment =>
          comment.location?.hunk &&
          isInlineLocation(comment.location) &&
          createHunkIdFromLocation(comment.location) === hunkId
      )
      .forEach(comment => {
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        const changeId = createChangeIdFromLocation(comment.location!);
        let lineComments = commentsByLine[changeId];
        if (!lineComments) {
          lineComments = [];
          commentsByLine[changeId] = lineComments;
        }
        lineComments.push(comment);
      });
```